util/util.go

```diff
@@ -447,3 +447,8 @@ func RetryOnConflictCause(fn func() (interface{}, error)) (obj interface{}, err
 	}
 	return nil, errors.Wrapf(err, "cannot finish API request due to too many conflicts")
 }
+
+func CouldFitDisk(replicaSize, storageScheduled, storageMaximum, storageReserved, storageAvailable, overProvisioningPercentage, minimalAvailablePercentage int64) bool {
```

Comment:
Put this function in the scheduler rather than util. It's the scheduler's algorithm.

Reply:
Also, pass an object instead of multiple values here. It's very easy to get the order messed up. And you can create a helper function to get the values from disk object, status object, and settings.